rakelib/docs.rake

```diff
@@ -342,3 +420,8 @@ end
 def format_links(text)
   text.gsub(/{([^}]+)}/, '[`\1`](#\1)')
 end
+
+def make_signature(function_name, params, return_type)
```

Comment:
I've personally found signatures with types a little hard to read, so I wanted to see what it would look like to just show how the function would be called and have the parameter list show what types are expected.

Reply:
This takes away pretty meaningful information from the signature, but we also have the parameters and types immediately below the signature. I'd advocate that between the parameter table and examples we can just remove the signature altogether. If we don't, I think we should leave them as they are.

Reply:
I think in many cases the _types_ aren't that useful, since they tend to be what you would expect, but the signature lets you quickly see the _order_ to pass the parameters. For instance, I personally can never remember if `file::write` is `file::write(filename, content)` or `file::write(content, filename)`, but I do know that both of those are strings.

Reply:
I don't think the current signatures (with the types) offer much value because they're so hard to parse. Compare the current `download_file` example:
```
download_file(String[1] $source, String[1] $destination, Boltlib::TargetSpec $targets, Optional[Hash[String[1], Any]] $options)
```
vs the updated version:
```
download_file($source, $destination, $targets, $description, $options) => ResultSet
```
The new version seems much more useful to me simply because it's easier to figure out what's going on. Granted, I don't use this stuff every day. If I did, it would probably be easier for me to parse that first example, but I think I'd still wonder why it's included if the relevant information for each parameter is in the table below. 

I do think those initial signatures have value though. I think a lot of more experienced users would only need to look at that first signature (and maybe a quick glance at the table) so know what they need to do. Generally in the docs, we try to follow a similar structure: basic syntax, reference info you need, examples.

Reply:
Ok, if other people find it useful we can leave it.